lib/bolt/applicator.rb

```diff
@@ -179,11 +230,37 @@ def count_statements(ast)
     def apply_ast(raw_ast, targets, options, plan_vars = {})
       ast = Puppet::Pops::Serialization::ToDataConverter.convert(raw_ast, rich_data: true, symbol_to_string: true)
 
+      # rubocop:disable Style/GlobalVars
+      if $future
+        %w[result resultset applyresult].each do |type|
+          Puppet.lookup(:loaders).private_environment_loader.load(:type, type)
```

Comment:
I don't _think_ this should be necessary. If we have any instances of these types to serialize, doesn't that imply the type must have been loaded?

Is there a specific case where omitting this causes a problem?